mailbox/store/src/main/java/org/apache/james/mailbox/store/PreDeletionHooks.java

```diff
@@ -44,6 +45,7 @@ public PreDeletionHooks(Set<PreDeletionHook> hooks) {
     public Mono<Void> runHooks(PreDeletionHook.DeleteOperation deleteOperation) {
         return Flux.fromIterable(hooks)
             .publishOn(Schedulers.elastic())
+            .limitRate(1)
```

Comment:
you could write a test with 2 hooks sharing a lock to check that:
the first hook take the lock and sleep for a second
the second hook try to take the lock and throws if it can't
WDYT?